openscenario/map_fragment/src/intersection.cpp

```diff
@@ -0,0 +1,254 @@
+// Copyright 2015 TIER IV, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <map_fragment/map_fragment.hpp>
+#include <map_fragment/road_segments/generate_lanelet_map.hpp>
+#include <rclcpp/rclcpp.hpp>
+
+enum IntersectionLeg { LEFT_LEG = 0, BOTTOM_LEG = 1, RIGHT_LEG = 2, TOP_LEG = 3 };
+
+auto getLegOrientation(IntersectionLeg leg) -> double { return leg * M_PI_2; }
+
+enum ConnectionType { LEFT_TURN = -1, FORWARD = 2, RIGHT_TURN = 1 };
+
+auto getConnectedLeg(IntersectionLeg leg, ConnectionType connection_type) -> IntersectionLeg
+{
+  return static_cast<IntersectionLeg>(((leg + connection_type) + 4) % 4);
+}
+
+auto main(const int argc, char const * const * const argv) -> int
+try {
+  using namespace map_fragment;
+  using namespace map_fragment::road_segments;
+
+  rclcpp::init(argc, argv);
+
+  auto node = rclcpp::Node(std::filesystem::path(argv[0]).stem());
+
+  const auto lane_width = [&]() {
+    node.declare_parameter("lane_width", default_value::width);
+    return node.get_parameter("lane_width").as_double();
+  }();
+
+  const auto leg_length = [&]() {
+    node.declare_parameter("leg_length", default_value::length);
+    return node.get_parameter("leg_length").as_double();
+  }();
+
+  const auto turn_radius = [&]() {
+    node.declare_parameter("turn_radius", default_value::turn_radius);
+    return node.get_parameter("turn_radius").as_double();
+  }();
```

Comment:
[Done](https://github.com/tier4/scenario_simulator_v2/pull/1124/commits/6c9501174370384f258595e143868496cc9672be).

Reply:
I tested the following commands and confirmed that an error occurs.

Command:
```
ros2 run map_fragment intersection --ros-args \
  -p number_of_lanes_going_in_all_directions:=2 \
  -p turn_radius:=10.0
```

Output:
```
Turn radius must not be smaller than half leg width. Actual values:
- Turn radius: 10.000000
- Half leg width: 20.000000
```

However, there is a problem in that the error message contains the term "half leg width," a term that one would not know what it means unless one knows the internal implementation. **I don't require this to be resolved in this pull request**, but when writing error messages, please do not use information that is not available to the user. This may be perceived by the user as a failure of an internal assertion rather than a failure based on the parameters given by the user.